### PR TITLE
[firebase-analytics] Remove native `setScreenName` method and deprecate `setCurrentScreen`

### PR DESCRIPTION
**Pull request by IjzerenHein**
# Why

The native `setScreenName/setCurrentScreen` methods are deprecated and have meanwhile been removed from the native SDKs. This makes it impossible to upgrade to the latest native Firebase SDKs and causes a warning with the current SDKs.

image

References:
- https://firebase.google.com/support/release-notes/ios#analytics_3
- https://firebase.google.com/docs/reference/android/com/google/firebase/analytics/FirebaseAnalytics#setCurrentScreen(android.app.Activity,%20java.lang.String,%20java.lang.String)
- https://firebase.google.com/support/release-notes/android#2020-08-14

This PR removes the no longer supported native-methods and implements a (temporary) JS version of `setCurrentScreen` which simply calls `logEvent('screen_view', { screen_name: 'foo' })`. This method is marked as deprecated and that it will be removed in SDK 45 to allow users some migration time.

Users need to replace `setCurrentScreen` with `logEvent('screen_view', { screen_name: 'foo' })`. 

image

# How

- Remove `setCurrentScreen` method from iOS/Android/web & pure JS implementation
- `setCurrentScreen` is marked as `@deprecated`
- Log one-time warning when calling `setCurrentScreen`
- Updated build files
- Updated tests
- Updated changelog 
- Update docs (`et gdad -p expo-firebase-analytics`)

# Test Plan

- All tests pass
- Updated docs (see screenshot)

### CODE REVIEW

File: packages/expo-firebase-analytics/src/Analytics.ts [hunk 1]
```diff
@@ -71,11 +71,16 @@ export async function setAnalyticsCollectionEnabled(isEnabled: boolean): Promise
...
+let hasWarnedAboutSetCurrentScreen = false;
 // @needsAudit
 /**
  * Sets the current screen name, which specifies the current visual context in your app. This helps
  * identify the areas in your app where users spend their time and how they interact with your app.
  *
+ * @deprecated FirebaseAnalytics.setCurrentScreen is deprecated.
+ * Use `logEvent('screen_view', { screen_name: 'MyScreen', ... })` instead.
+ * This API will be removed in SDK 45.
+
```
Comment (Simek) [inline packages/expo-firebase-analytics/src/Analytics.ts:83]:
```suggestion
 * @deprecated __Deprecated.__ Use [`logEvent('screen_view', { screen_name: 'MyScreen', ... })`](#logevent)
 * instead. This API will be removed in SDK 45.
 *
```
I will probably need to add some standardization for the comments from `@deprecated` annotation and apply the changes retrospectively. I think that the "Deprecated." message should be added automatically before the comment content. 😅 

Also I have added link to the recommended migration method.

Comment (IjzerenHein) [reply packages/expo-firebase-analytics/src/Analytics.ts:83]: Awesome, thanks for that 😃

**Review by Simek — APPROVED** [review]
LGTM form the implementation standpoint, but it would be nice if someone with more experience in API management can confirm that removal here is fine. 😉

**Comment by jeff-polls** [comment]
Great job guys. I appreciate that you left the JS method in there. What's the reasoning behind ever removing it?
I imagine most devs will then create their own identical JS method that wraps logEvent

**Pull request merged**